sql/commitment.sql

```diff
@@ -0,0 +1,375 @@
+		allocate_citycost numeric,
+		allocate_nccstate numeric,
+		allocate_nccfederal numeric,
+		allocate_nccother numeric,
+		allocate_noncitycost numeric,
+		allocate_total numeric,
+		commit_ccnonexempt numeric,
+		commit_ccexempt numeric,
+		commit_citycost numeric,
+		commit_nccstate numeric,
+		commit_nccfederal numeric,
+		commit_nccother numeric,
+		commit_noncitycost numeric,
+		commit_total numeric,
+		spent_ccnonexempt numeric,
+		spent_ccexempt numeric,
+		spent_citycost numeric,
+		spent_nccstate numeric,
+		spent_nccfederal numeric,
+		spent_nccother numeric,
+		spent_noncitycost numeric,
+		spent_total numeric,
+		spent_total_checkbooknyc numeric
+	);
+	
+COMMIT;
+
+-- Populate source tables
+BEGIN;
+	COPY project_source FROM 'cpdb_projects_23adopt.csv' DELIMITER ',' CSV HEADER;
+COMMIT;
+
+BEGIN;
+	COPY commitment_source FROM 'cpdb_planned_commitments_23adopt.csv' DELIMITER ',' CSV HEADER;
+COMMIT;
+
+-- Structure of target tables
+-- Right now, we are using data as primary keys. We should evaluate when they should be replaced with separate ids
+BEGIN;
+	CREATE TABLE managing_code (
+		id char(3) PRIMARY KEY
+	);
+
+	-- Should likely create an id distinct from initials
+	CREATE TABLE agency (
+		initials text PRIMARY KEY,
+		name text
+	);
+	
+ -- likely unnecessary as a table because the project table already captures both its managing code and its managing agency 
+	-- CREATE TABLE agency_managing_code (
+	-- 	agency_initials text references agency(initials),
+	-- 	managing_code_id char(3) references managing_code(id)
+	-- );
+	
+	CREATE TYPE project_category AS ENUM (
+		'Fixed Asset',
+		'Lump Sum',
+		'ITT, Vehicles, and Equipment'
+	);
+		
+	CREATE TABLE project (
+		PRIMARY KEY (managing_code, id),
+    managing_code char(3) references managing_code(id),
+		id text,
+		managing_agency text references agency(initials),
+		description text,
+		min_date date,
+		max_date date,
+		category project_category,
+		adopt_city_non_exempt numeric,
+		adopt_city_exempt numeric,
+		adopt_city_total numeric,
+		adopt_state numeric,
+		adopt_federal numeric,
+		adopt_other numeric,
+		adopt_non_city numeric,
+		adopt_total numeric,
+		allocate_city_non_exempt numeric,
+		allocate_city_exempt numeric,
+		allocate_city_total numeric,
+		allocate_state numeric,
+		allocate_federal numeric,
+		allocate_other numeric,
+		allocate_non_city numeric,
+		allocate_total numeric,
+		commit_city_non_exempt numeric,
+		commit_city_exempt numeric,
+		commit_city_total numeric,
+		commit_state numeric,
+		commit_federal numeric,
+		commit_other numeric,
+		commit_non_city numeric,
+		commit_total numeric,
+		spent_city_non_exempt numeric,
+		spent_city_exempt numeric,
+		spent_city_total numeric,
+		spent_state numeric,
+		spent_federal numeric,
+		spent_other numeric,
+		spent_non_city numeric,
+		spent_total numeric,
+		checkbook_nyc numeric		
+	);
+	
+	-- Should likely create an id separate from version
+	CREATE TABLE capital_commitment_plan(
```

Comment:
Would dropping this table and relation break any other tables? I'm pretty confident we don't actually need it for this initial build. I appreciate the thought but it's getting into the question of modeling multiple years of CPDB which is out of scope for this build.

Reply:
I'll drop it for the next version of the erd